### PR TITLE
Fix for cellphone exploit 

### PR DESCRIPTION
**Pull request by Wackbatt**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #<issue ID here>. <If applicable.>

#### Changes proposed in this pull request: 
- Addition of a character limit for player to player, player to ems/cop/admins messages to prevent client and server instabilities and possible crashes.

### CODE REVIEW

**Comment by SimZor** [comment]
Ugh....no spaces :(

also why """""""""""""""""""" when you can just ''?

File: life_server/functions.sqf [hunk 1]
```diff
@@ -130,6 +130,9 @@ compileFinal "
 private [""_msg"",""_to""];
     ctrlShow[3022,false];
     _msg = ctrlText 3003;
+    _length = count (toArray(_msg));
+    if(_msg == """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3022,true];};
```
Comment (Jawshy) [inline life_server/functions.sqf]: `if (`* - Whitespace.

File: life_server/functions.sqf [hunk 2]
```diff
@@ -144,6 +147,9 @@ compileFinal "
     private [""_msg"",""_to""];
     ctrlShow[3015,false];
     _msg = ctrlText 3003;
+    _length = count (toArray(_msg));
+    if(_msg == """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3015,true];};
```
Comment (Jawshy) [inline life_server/functions.sqf]: `if (`* - Whitespace.

File: life_server/functions.sqf [hunk 3]
```diff
@@ -163,6 +169,9 @@ compileFinal "
     _msg = ctrlText 3003;
     _to = ""The Police"";
     if (_msg isEqualTo """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3016,true];};
+    _length = count (toArray(_msg));
+    if(_msg == """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3016,true];};
```
Comment (Jawshy) [inline life_server/functions.sqf]: `if (`* - Whitespace.

File: life_server/functions.sqf [hunk 4]
```diff
@@ -177,6 +186,9 @@ compileFinal "
     _msg = ctrlText 3003;
     _to = ""The Admins"";
     if (_msg isEqualTo """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3017,true];};
+    _length = count (toArray(_msg));
+    if(_msg == """") exitWith {hint ""You must enter a message to send!"";ctrlShow[3017,true];};
```
Comment (Jawshy) [inline life_server/functions.sqf]: `if (`* - Whitespace.

**Comment by WolfSkin1337** [comment]
And What is with `==` to `isEqualTo`

**Comment by BoGuu** [comment]
Maybe a localised string? Then LGTM

File: life_server/functions.sqf [hunk 5]
```diff
@@ -130,6 +130,8 @@ compileFinal "
 private [""_msg"",""_to""];
     ctrlShow[3022,false];
     _msg = ctrlText 3003;
+    _length = count (toArray(_msg));
+    if (_length > 400) exitWith {hint ""Your message cannot exceed 400 characters"";ctrlShow[3022,true];};
     _to = ""EMS Units"";
```
Comment (DarkSilencerZA) [inline life_server/functions.sqf]: Consistency.. move it down.

**Comment by WolfSkin1337** [comment]
@Wackbatt 
see https://github.com/AsYetUntitled/Framework/blob/master/Altis_Life.Altis/stringtable.xml#L1282

**Comment by trinitynet** [comment]
i corrected the conflict, tho the message length sting needs to be localised

**Comment by BoGuu** [comment]
LGTM

**Comment by WolfSkin1337** [comment]
LGTM but what about the localized string for the Test: `Your message cannot exceed 400 characters` ?

**Comment by BoGuu** [comment]
@WolfSkin1337 7feaf52b91821339eed1abb9a952f2acc480d0d2

**Comment by WolfSkin1337** [comment]
nope there isnt a string for that

File: life_server/functions.sqf [hunk 6]
```diff
@@ -130,6 +130,8 @@ compileFinal "
 private [""_msg"",""_to""];
     ctrlShow[3022,false];
     _msg = ctrlText 3003;
+    _length = count (toArray(_msg));
+    if (_length > 400) exitWith {hint ""Your message cannot exceed 400 characters"";ctrlShow[3022,true];};
```
Comment (WolfSkin1337) [inline life_server/functions.sqf]: `Your message cannot exceed 400 characters`
to
`STR_CELLMSG_Over400`

File: life_server/functions.sqf [hunk 7]
```diff
@@ -144,7 +146,11 @@ compileFinal "
     private [""_msg"",""_to""];
     ctrlShow[3015,false];
     _msg = ctrlText 3003;
+
+    _length = count (toArray(_msg));
+    if (_length > 400) exitWith {hint ""Your message cannot exceed 400 characters"";ctrlShow[3015,true];};
```
Comment (WolfSkin1337) [inline life_server/functions.sqf]: `Your message cannot exceed 400 characters`
to
`STR_CELLMSG_Over400`

File: life_server/functions.sqf [hunk 8]
```diff
@@ -162,7 +168,10 @@ compileFinal "
     ctrlShow[3016,false];
     _msg = ctrlText 3003;
     _to = ""The Police"";
+
     if (_msg isEqualTo """") exitWith {hint localize ""STR_CELLMSG_EnterMSG"";ctrlShow[3016,true];};
+    _length = count (toArray(_msg));
+    if (_length > 400) exitWith {hint ""Your message cannot exceed 400 characters"";ctrlShow[3016,true];};
```
Comment (WolfSkin1337) [inline life_server/functions.sqf]: `Your message cannot exceed 400 characters`
to
`STR_CELLMSG_Over400`

File: life_server/functions.sqf [hunk 9]
```diff
@@ -176,7 +185,10 @@ compileFinal "
     ctrlShow[3017,false];
     _msg = ctrlText 3003;
     _to = ""The Admins"";
+
     if (_msg isEqualTo """") exitWith {hint localize ""STR_CELLMSG_EnterMSG"";ctrlShow[3017,true];};
+    _length = count (toArray(_msg));
+    if (_length > 400) exitWith {hint ""Your message cannot exceed 400 characters"";ctrlShow[3017,true];};
```
Comment (WolfSkin1337) [inline life_server/functions.sqf]: `Your message cannot exceed 400 characters`
to
`STR_CELLMSG_Over400`

**Comment by trinitynet** [comment]
added localisations from @WolfSkin1337 table

**Comment by WolfSkin1337** [comment]
LGTM

**Comment by BoGuu** [comment]
Ready to merge @Jawshy

**Pull request merged**